pkg/kubectl/cmd/set/set_selector_test.go

```diff
@@ -46,14 +46,14 @@ func TestUpdateSelectorForObjectTypes(t *testing.T) {
 			},
 		}}
 
-	rc := api.ReplicationController{}
-	ser := api.Service{}
-	dep := extensions.Deployment{Spec: extensions.DeploymentSpec{Selector: &before}}
-	ds := extensions.DaemonSet{Spec: extensions.DaemonSetSpec{Selector: &before}}
-	rs := extensions.ReplicaSet{Spec: extensions.ReplicaSetSpec{Selector: &before}}
-	job := batch.Job{Spec: batch.JobSpec{Selector: &before}}
-	pvc := api.PersistentVolumeClaim{Spec: api.PersistentVolumeClaimSpec{Selector: &before}}
-	sa := api.ServiceAccount{}
+	rc := v1.ReplicationController{}
+	ser := v1.Service{}
+	dep := extensionsv1beta1.Deployment{Spec: extensionsv1beta1.DeploymentSpec{Selector: &before}}
```

Comment:
for objects which exist in multiple versions, we should ensure the various versions work properly

Reply:
```
	switch t := obj.(type) {
	case *v1.Service:
		t.Spec.Selector, err = copyOldSelector()
	default:
		err = fmt.Errorf("setting a selector is only supported for Services")
	}
```
`set selector` now just support `v1.service`.